### PR TITLE
Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by oliviertassinari**
Benchmark:

- https://www.djangoproject.com/foundation/cla/
- https://code.facebook.com/cla/

### CODE REVIEW

**Comment by oliviertassinari** [comment]
After further discussion with @mbrookes we might want to try a different strategy on this one.
We could ask each contributors to agree to license the content as MIT. This would be reduce unknown, reduce the fiction to contributing. From what I understand, once the code is licensed we can distribute it under a commercial license. All we would need to do it include the copyright and license notice.

On a side note, almost all the XGrid code is released as MIT under the `@material-ui/data-grid` package. It will be important to change that, both for IP and bundle size considerations.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Contributing to Material-UI X
+
+## Material-UI X vs. Material-UI
+
+Material-UI X is an extension of Material-UI.
+The repositories are part of the same codebase.
+`mui-org/material-ui-x` imports the code infrastructure from [`mui-org/material-ui`](https://github.com/mui-org/material-ui).
+You can find the contribution guide of the main repository at [mui-org/material-ui/CONTRIBUTING.md](https://github.com/mui-org/material-ui/blob/HEAD/CONTRIBUTING.md).
+
+Most of the content on the contribution guide of Material-UI applies to this repository.
```
Comment (mbrookes) [inline CONTRIBUTING.md:10]:
```suggestion
Most of the content in the Material-UI "contributing" guide applies to this repository.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Contributing to Material-UI X
+
+## Material-UI X vs. Material-UI
+
+Material-UI X is an extension of Material-UI.
+The repositories are part of the same codebase.
+`mui-org/material-ui-x` imports the code infrastructure from [`mui-org/material-ui`](https://github.com/mui-org/material-ui).
+You can find the contribution guide of the main repository at [mui-org/material-ui/CONTRIBUTING.md](https://github.com/mui-org/material-ui/blob/HEAD/CONTRIBUTING.md).
```
Comment (mbrookes) [inline CONTRIBUTING.md:8]:
```suggestion
You can find the "contributing" guide for the main repository in [mui-org/material-ui/CONTRIBUTING.md](https://github.com/mui-org/material-ui/blob/HEAD/CONTRIBUTING.md).
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Contributing to Material-UI X
+
+## Material-UI X vs. Material-UI
+
+Material-UI X is an extension of Material-UI.
+The repositories are part of the same codebase.
```
Comment (mbrookes) [inline CONTRIBUTING.md:6]: What does this mean?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+`mui-org/material-ui-x` imports the code infrastructure from [`mui-org/material-ui`](https://github.com/mui-org/material-ui).
+You can find the contribution guide of the main repository at [mui-org/material-ui/CONTRIBUTING.md](https://github.com/mui-org/material-ui/blob/HEAD/CONTRIBUTING.md).
+
+Most of the content on the contribution guide of Material-UI applies to this repository.
+However, there are a few differences:
```
Comment (mbrookes) [inline CONTRIBUTING.md:11]:
```suggestion
There are, however,  a few differences:
```

**Review by mbrookes — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 3 above)
Comment (oliviertassinari) [reply CONTRIBUTING.md:6]: Same problem, same solution

**Pull request merged**